test/test_preserve_bundler_env.rb

```diff
@@ -0,0 +1,33 @@
+require_relative "helper"
+require_relative "helpers/integration"
+
+class TestPreserveBundlerEnv < TestIntegration
+  def setup
+    skip NO_FORK_MSG unless HAS_FORK
+    super
+  end
+
+  # It does not wipe out BUNDLE_GEMFILE et al
+  def test_usr2_restart_preserves_bundler_environment
```

Comment:
I thought about the skip here more and I guess it makes sense that while the code technically _runs_ on non-usr2 restarts it doesn't actually have a testable effect unless you're doing a usr2 restart. This test should be skipped if platform support isn't present (the way you had it before).